[choonzies] iP

choonzies opened the pull request:
# Primo
El Primo will keep track of your tasks using a CLI interface!
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

El Primo is
- Friendly
- Easy-going
- Mexican
- ~~FOR PAIN AND FOR GLORY!~~

3 Types of Tasks management supported!
1. Todo Tasks
2. Deadline Tasks
3. Event Tasks

And it is **_FREE_**_!!_

Features available:
- [x] Mark and unmark tasks
- [x] Add new tasks
- [x] Show list of tasks
- [x] Find task by name
- [ ] Win Bounty matches :neutral_face:

Sample commands:
`todo shopping`
`deadline homeword /by 2024/09/07`
`event orientation /from 2024-09-04 /to 2024-09-11`

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Primo("data/tasks.txt").run(); // Runs the application with the specified file path
}
```

![El Primo](https://static.wikia.nocookie.net/brawlstars/images/0/04/El_Primo_Skin-Default.png/revision/latest?cb=20200225131129)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

choonzies committed:
Add skeletal version with new chatbot name

choonzies committed:
Add echo feature with UI updates

- Echoes command by user
- Closes conversation for bye command
- Removed dashes, added name with colon format instead

choonzies committed:
Add list variable with list printing method

choonzies committed:
Include 'added' string and list storing and retrieval feature

choonzies committed:
Add Task class

choonzies committed:
Modify print statements to include details of list command

choonzies committed:
Add mark and unmark commands

choonzies committed:
Create classes for ToDo, Deadline, Event

choonzies committed:
Add commands for todo, deadline, event and adjust text strings

choonzies committed:
Implement automated Text UI Testing

choonzies committed:
Add exception handling for:
- Invalid command word
- Invalid structure of commands
- Empty descriptions or parameters

choonzies committed:
Add delete command feature

choonzies committed:
Convert logic conditionals to the use of Java enums instead

choonzies committed:
Fix index of substrings for exception handling

choonzies committed:
Add method readData to read from data.txt

choonzies committed:
Add file corruption handling

choonzies committed:
Change data.txt format and parsing sequence

choonzies committed:
Add getList() method which will return written data

choonzies committed:
Add file writing when list changes

choonzies committed:
Merge branch 'branch-Level-7'

choonzies committed:
Add intepretation for LocalDate input

choonzies committed:
Add DateTimeFormatter for date outputs

choonzies committed:
Merge branch 'branch-Level-8'

choonzies committed:
Create OOP files with todo fields and methods

choonzies committed:
Implement OOP style to codebase

choonzies committed:
Remove commented old code and edit tests

choonzies committed:
Remove old unused code

choonzies committed:
Merge branch 'A-MoreOOP'

choonzies committed:
Sort the java files into respective packages

choonzies committed:
Merge branch 'A-Packages'

choonzies committed:
Merge branch 'add-gradle-support'

choonzies committed:
Add gradle and checkstyle

choonzies committed:
Merge branch 'A-Gradle'

choonzies committed:
Add Junit tests for parser and command

choonzies committed:
Merge branch 'A-JUnit'

choonzies committed:
Create a .jar file in build/libs/Primo.jar

choonzies committed:
Merge branch 'A-Jar'

choonzies committed:
Add JavaDoc to all except tasks

choonzies committed:
Edit minor segments to fit the coding standard

choonzies committed:
Add find command that finds keyword

choonzies committed:
Add find command that finds keyword

choonzies committed:
Add JavaDoc for new FindCommand file

choonzies committed:
Added JavaDoc for new FindCommand file

choonzies committed:
Merge branch 'branch-A-JavaDoc'

choonzies committed:
Merge branch 'branch-A-CodingStandard'

choonzies committed:
Merge branch 'branch-level-9'

calebyyy commented on src/main/java/Primo.java:
Good variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/Primo.java:
Good variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/ByeCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/Command.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/DeadlineCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/DeleteCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/EventCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/FindCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/ListCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/MarkCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/TodoCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/commands/UnmarkCommand.java:
Good boolean variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/exception/PrimoException.java:
Good JavaDocs.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/parser/Parser.java:
Good variable names.

choonzies replied:
Thanks!

calebyyy commented on src/main/java/storage/Storage.java:
Shouldnt use wildcard commands.

choonzies replied:
Good catch, thanks!

calebyyy reviewed (commented):
Looks good. Be careful of wildcard imports.

choonzies committed:
Remove wildcard imports in Storage.java

choonzies committed:
Fix bug relating to creation of data files at first run

choonzies committed:
Use checkstyle and fix relevant style errors

choonzies committed:
Merge branch 'A-CheckStyle'

tyxiangs commented on src/main/java/Primo.java:
```diff
@@ -0,0 +1,78 @@
+            Files.createDirectories(directoryPath);
+            Files.createFile(filePath);
+            ui.showLoadingError();
+            tasks = new TaskList(); // Initializes an empty task list on error
+            this.tasks = new TaskList();
```
This looks like duplicate initialization. You may consider deleting one of them. Ignore me if you need it.

tyxiangs commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,188 @@
+            if (eventDescription.isEmpty()) {
+                throw new PrimoException("Description cannot be empty! Expected deadline <string> /by <string>");
+            }
+            String from = fullCommand.substring(eventToIndex + 5, eventFinalIndex).trim();
```
I think you can use a more specific name like fromdate or fromDateTime

tyxiangs commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,118 @@
+                String name = "";
+                String deadline = "";
+                String from = "";
+                String to = "";
```
I think you can use a more specific name like fromdate or fromDateTime or toDate

tyxiangs commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,34 @@
+
+public class EventTask extends Task {
+    private static final String SYMBOL = "E";
+    private String from;
+    private String to;
```
I think you can use a more specific name like fromDate

tyxiangs reviewed (commented):
Overall, a really good job done 👍
I just highlighted a few minor changes which could be reviewed.

choonzies committed:
Add javafx tutorial files

choonzies committed:
Integrate GUI with commands

choonzies committed:
Add icon, title, customise images

choonzies committed:
Merge branch 'Level-10'

choonzies committed:
Touch up UI

choonzies committed:
Merge branch 'branch-Level-10'

choonzies committed:
Add assertions to ensure code correctness

Current code has no use of assertions to check programmer's errors.

Programmer's errors may be present.

Let's add assertions to ensure code flow correctness, especially in
control flow statements in parser.

choonzies committed:
Apply SLAP to enhance code quality

choonzies committed:
Merge pull request #2 from choonzies/A-CodeQuality

Apply SLAP to enhance code quality

choonzies committed:
Merge branch 'master' into A-Assertions

choonzies committed:
Merge pull request #1 from choonzies/A-Assertions

Add assertions to ensure code correctness

choonzies committed:
Add gradle.yml for CI workflow

choonzies committed:
Merge branch 'A-CI'

choonzies committed:
Add note feature for todo tasks and adjust parsing accordingly

choonzies committed:
Add note feature for deadline tasks and adjust parsing accordingly

choonzies committed:
Add note feature for event tasks and adjust parsing accordingly

choonzies committed:
Fix bug regarding error reading data with >=10 tasks

choonzies committed:
Merge branch 'D-Notes'

choonzies committed:
Anchor text field, send and scrollpane

choonzies committed:
Set fixed window size

choonzies committed:
Add CSS to enhance Gui

choonzies committed:
Merge branch 'A-BetterGui'

choonzies committed:
Add voicelines for Primo personality

choonzies committed:
Change title to El Primo

choonzies committed:
Merge branch 'A-Personality'

choonzies committed:
Handle error for invalid commands

choonzies committed:
Merge branch 'A-MoreErrorHandling'

choonzies committed:
Add user guide

choonzies committed:
Fix typos

choonzies committed:
Merge branch 'A-UserGuide'

choonzies committed:
Add more Javadoc and JUnit tests

choonzies committed:
Update README.md
